### PR TITLE
Support timeToClose option

### PR DESCRIPTION
**Pull request by agix**

### CODE REVIEW

File: src/actions/OptionsActions.js [hunk 1]
```diff
@@ -90,6 +92,17 @@ class OptionsActions {
 
     return this.deactivateShortLogin();
   }
+
+  changeDelay(delay) {
+    secretin.editOption('timeToClose', delay)
+      .then(() => {
+        this.changeDelaySuccess(secretin.currentUser.options.timeToClose);
```
Comment (Calyhre) [inline src/actions/OptionsActions.js]: You should always pass objects to actions.
```javascript
this.changeDelaySuccess({
  timeToClose: secretin.currentUser.options.timeToClose,
})
```

File: src/components/App.js [hunk 2]
```diff
@@ -39,7 +39,7 @@ class App extends Component {
   render() {
     return (
       <Router basename={process.env.PUBLIC_URL}>
-        <div className="App">
+        <div className="App" >
```
Comment (Calyhre) [inline src/components/App.js]: Nope

File: src/stores/AppUIStore.js [hunk 3]
```diff
@@ -20,6 +20,27 @@ class AppUIStore {
     this.state = new AppUIState({
       savedUsername: secretin.getSavedUsername(),
     });
+
+    this.disconnectingEvent = null;
+
+    window.addEventListener('focus', () => { this.cameBack(); });
+    window.addEventListener('blur', () => { this.left(); });
```
Comment (Calyhre) [inline src/stores/AppUIStore.js]: No way, those should be actions, called by a component.

File: src/stores/AppUIStore.js [hunk 4]
```diff
@@ -30,6 +51,19 @@ class AppUIStore {
     );
   }
 
+  disconnectUser() {
+    secretin.currentUser.disconnect();
```
Comment (Calyhre) [inline src/stores/AppUIStore.js]: That should be in an action.

File: src/actions/AppUIActions.js [hunk 5]
```diff
@@ -15,10 +15,15 @@ class AppUIActions {
...
       'loginUserSuccess',
       'loginUserFailure',
-      'appReady'
+      'appReady',
     );
   }
 
+  disconnectUser() {
+    secretin.currentUser.disconnect();
+    return true;
```
Comment (Calyhre) [inline src/actions/AppUIActions.js]: Why not returning directly `secretin.currentUser.disconnect` ?

File: src/actions/OptionsActions.js [hunk 6]
```diff
@@ -90,6 +92,17 @@ class OptionsActions {
...
+
+  changeDelay(delay) {
+    secretin.editOption('timeToClose', delay)
+      .then(() => {
+        this.changeDelaySuccess({ timeToClose: secretin.currentUser.options.timeToClose });
+      })
+      .catch(() => {
+        this.changeDelayFailure();
+      });
+    return true;
```
Comment (Calyhre) [inline src/actions/OptionsActions.js]: Same, you should return the promise here.

File: src/components/options/OptionsContainer.js [hunk 7]
```diff
@@ -29,6 +28,58 @@ class OptionsContainer extends Component {
...
+
+    this.handleChangeDelay = this.handleChangeDelay.bind(this);
+    this.handleClickMinus = this.handleClickMinus.bind(this);
+    this.handleClickPlus = this.handleClickPlus.bind(this);
+    this.confirmChangeDelay = this.confirmChangeDelay.bind(this);
+
+    this.confirmDelayTimeout = null;
+
+    this.state = {
+      delay: props.options.get('timeToClose') ? props.options.get('timeToClose') : 30,
```
Comment (Calyhre) [inline src/components/options/OptionsContainer.js]: Weird to show 30 for someone who does not have set this option, and so does not have the disconnect enabled.
cf: https://github.com/secretin/secretin-app/pull/22/files#diff-57aa11691094c3295e1f9dcbe8432ba2R61

File: src/actions/AppUIActions.js (hunk 5 above)
Comment (agix) [inline src/actions/AppUIActions.js]: disconnect doesn't return anything neither but if it's better this way.

File: src/actions/OptionsActions.js (hunk 6 above)
Comment (agix) [inline src/actions/OptionsActions.js]: ok

File: src/components/options/OptionsContainer.js (hunk 7 above)
Comment (agix) [inline src/components/options/OptionsContainer.js]: This feature is enabled from the beginning (sleeping feature) https://github.com/secretin/secretin-lib/blob/master/src/User.js#L327

File: src/actions/AppUIActions.js (hunk 5 above)
Comment (agix) [inline src/actions/AppUIActions.js]: We should use the dispatch stuff from alt

File: src/components/App.js [hunk 8]
```diff
@@ -36,6 +39,32 @@ class App extends Component {
...
+    this.disconnectingEvent = null;
+
+    this.cameBack = this.cameBack.bind(this);
+    this.left = this.left.bind(this);
+
+    window.addEventListener('focus', this.cameBack);
+    window.addEventListener('blur', this.left);
+  }
+
+  cameBack() {
```
Comment (Calyhre) [inline src/components/App.js]: Renommer en `onCameBack` et `onLeft` ?

**Pull request merged**